src/plugins/fakeProfileThemesAndEffects/index.tsx

```diff
@@ -0,0 +1,496 @@
+                        if (base125StrToBase10CSSColor(decoded3y3[0]) === -2)
+                            updateUserEffectID(user, base125StrToBase10ItemID(decoded3y3[1]));
+                        else
+                            updateUserEffectID(user, base125StrToBase10ItemID(decoded3y3[2]));
+                    }
+                    return user;
+                } else if(user.profileEffectID !== undefined) {
+                    const decoded3y3: [string, string, string] = decode3y3(user.bio);
+                    const primaryColor: number = base125StrToBase10CSSColor(decoded3y3[0]);
+                    if (primaryColor === -2)
+                        legacy3y3(user, decoded3y3[0]);
+                    else
+                        updateUserThemeColors(user, primaryColor, base125StrToBase10CSSColor(decoded3y3[1]));
+                    return user;
+                }
+            }
+            const decoded3y3: [string, string, string] = decode3y3(user.bio);
+            const primaryColor: number = base125StrToBase10CSSColor(decoded3y3[0]);
+            if (primaryColor === -2) {
+                legacy3y3(user, decoded3y3[0]);
+                updateUserEffectID(user, base125StrToBase10ItemID(decoded3y3[1]));
+            } else {
+                updateUserThemeColors(user, primaryColor, base125StrToBase10CSSColor(decoded3y3[1]));
+                updateUserEffectID(user, base125StrToBase10ItemID(decoded3y3[2]));
+            }
+        }
+        return user;
+    },
+    add3y3Builder(): JSX.Element {
+        if (settings.store["Hide 3y3 Builder"] === true) return <></>;
+
+        const [primaryColor, setPrimaryColor]: [number, (v: number) => void] = useState(-1);
+        const [accentColor, setAccentColor]: [number, (v: number) => void] = useState(-1);
+        const [effectID, setEffectID]: [string, (v: string) => void] = useState("");
+        const [effectName, setEffectName]: [string, (v: string) => void] = useState("");
+        const [buildLegacy3y3, setBuildLegacy3y3]: [boolean, (v: boolean) => void] = useState(false);
+
+        return (
+            <>
+                <Text
+                    tag={"h3"}
+                    variant={"eyebrow"}
+                    style={{display: "inline"}}
+                >
+                    {"3y3 Builder"}
+                </Text>
+                <Button
+                    look={Button.Looks.LINK}
+                    color={Button.Colors.PRIMARY}
+                    size={Button.Sizes.TINY}
+                    style={{
+                        display: primaryColor === -1 && accentColor === -1 && effectID === "" ? "none" : "inline",
+                        height: "14px",
+                        marginLeft: "4px",
+                        minHeight: "0",
+                        paddingBottom: "0",
+                        paddingTop: "0"
+                    }}
+                    onClick={(): void => {

```

Comment:
Not everything needs types

```suggestion
                    onClick={() => {
```